workers/unity/Assets/Gdk/Core/Tests/Editmode/Worker/WorkerBaseTests.cs

```diff
@@ -6,31 +7,22 @@ namespace Improbable.Gdk.Core.EditmodeTests
     [TestFixture]
     public class WorkerBaseTests
     {
-        internal class UnityTestWorker : WorkerBase
-        {
-            public const string WorkerType = "UnityTestWorker";
-
-            public override string GetWorkerType => WorkerType;
-
-            public UnityTestWorker(string workerId, Vector3 origin) : base(workerId, origin)
-            {
-            }
-        }
-
         [Test]
         public void WorkerBase_should_throw_exception_when_WorkerId_is_null_or_empty()
         {
             // Empty check
             var empty_exception = Assert.Throws<System.ArgumentException>(() =>
             {
                 var worker = new UnityTestWorker("", new Vector3());
```

Comment:
Surround with a `using`? `Dispose` will never be called in this case (and below)...

Reply:
UnityTestWorker constructor will not finish in this case so the Dispose should actually not get called. And unfortunately the `using` block would still not make it get called because it would get compiled to this:

```cs
UnityTestWorker unityTestWorker = new UnityTestWorker("", new Vector3());
try
{
}
finally
{
    if (unityTestWorker != null)
    {
        ((IDisposable)unityTestWorker).Dispose();
    }
}
```